hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/db/RocksDatabase.java

```diff
@@ -979,9 +982,17 @@ public void deleteFilesNotMatchingPrefix(
                   + " {} from db: {}", sstFileName,
               liveFileMetaData.columnFamilyName(), db.get().getName());
           db.get().deleteFile(sstFileName);
+          filesToBeDeleted.add(new File(liveFileMetaData.path(),
+              liveFileMetaData.fileName()));
         }
       }
     }
+
+    Iterator<File> files = filesToBeDeleted.iterator();
+    while (files.hasNext()) {
```

Comment:
Why do we need to wait for file deletion? Once manifest is updated, it doesn't matter if file exist or got deleted.

Reply:
The deletes are asynchronous so we need the deletes to be synchronous here.

Reply:
Manifest files get updated only when the deletes are processed.

Reply:
I see. 

If delete to be synchronous, I think `waitForFileDelete` should be internal to `RocksDatabase` or should be moved to [RocksDBUtils`](https://github.com/apache/ozone/blob/master/hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/RocksDBUtils.java) or [RdbUtil](https://github.com/apache/ozone/blob/master/hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/rocksdb/util/RdbUtil.java). As the name says `RDBCheckpointUtils` is for Checkpoint and not for a general RDB util.

Reply:
done